pkg/3scale/amp/operator/upgrade.go

```diff
@@ -389,3 +394,60 @@ func (u *UpgradeApiManager) upgradeAPIManagerCRDatabaseDefaults() (reconcile.Res
 
 	return reconcile.Result{}, nil
 }
+
+func (u *UpgradeApiManager) upgradeSystemS3() (reconcile.Result, error) {
```

Comment:
With the current implementation, we need to upgrade the system-environment ConfigMap to include the new S3 related fields with their default values (the empty string). We also need to update the environment of the containers that are supposed to use the S3 configuration